src/libopensc/card-entersafe.c

```diff
@@ -220,20 +222,27 @@ static int entersafe_cipher_apdu(sc_card_t *card, sc_apdu_t *apdu,
 		 LOG_FUNC_RETURN(card->ctx, SC_ERROR_OUT_OF_MEMORY);
 	 EVP_CIPHER_CTX_set_padding(ctx,0);
 
-	 if(keylen == 8)
-		  EVP_EncryptInit_ex(ctx, EVP_des_ecb(), NULL, key, iv);
-	 else if (keylen == 16) 
-		  EVP_EncryptInit_ex(ctx, EVP_des_ede(), NULL, key, iv);
-	 else
+	if (keylen == 8) {
+	 	alg = sc_evp_cipher(card->ctx, "DES-ECB");
+		EVP_EncryptInit_ex(ctx, alg, NULL, key, iv);
+	} else if (keylen == 16) {
+	 	alg = sc_evp_cipher(card->ctx, "DES-EDE");
+		EVP_EncryptInit_ex(ctx, alg, NULL, key, iv);
```

Comment:
maybe one more nit that the the line `EVP_EncryptInit_ex(ctx, alg, NULL, key, iv);` is same in both of the condition so it can be moved out to avoid code duplication (similarly as in the previously mentioned case)

Reply:
TBH the quality of the code of this card is soo poor I did not want to change the structure at all. It doesn't even do error checking, and I already fixed a memory leak here ... sounds like the code of this driver could use an overhaul.
I'd prefer to leave as is.

Reply:
np. Lets keep it this way